src/mixins/view.js

```diff
@@ -163,11 +167,18 @@ export default {
     // https://github.com/jashkenas/backbone/blob/1.2.3/backbone.js#L1235
     this._removeElement();
 
+    if (shouldTriggerDetach) {
+      this._isAttached = false;
+      this.triggerMethod('detach', this);
+    }
+
     // remove children after the remove to prevent extra paints
     this._removeChildren();
 
     this._destroyBehaviors(args);
 
+    this._isDestroyed = true;
```

Comment:
I see what you're saying... that would be a problem.  I wasn't thinking in terms of side effects during children/behavior destruction, I was thinking in terms of our pattern of triggering events like attach, render, and destroy.  Especially after this PR, we are setting a flag _immediately_ before triggering a corresponding event in every case.  In other words, at the instant we change the state of an object (rendered, destroyed, attached) we also announce that fact with an event.

The question as I see it is, when exactly the view is "destroyed" as a semantic concept?  That could be before or after destroying either children, behaviors, or both.  Attempting to answer that question myself, I would have a hard time considered a view "destroyed" while pieces of it (children, behaviors) are still "alive."  I'd rather leave both flag and fire at the end method because I think the pattern is more important than protecting against side effects of a behavior or child destroy, which is probably a symptom of bad app design.